turbo/src/main/kotlin/dev/hotwire/turbo/delegates/TurboWebFragmentDelegate.kt

```diff
@@ -349,13 +350,13 @@ internal class TurboWebFragmentDelegate(
             else -> visitOptions
         }
 
-        viewLifecycleOwner.lifecycleScope.launch {
+        viewLifecycleOwner?.lifecycleScope?.launch {
             val snapshot = when (options.action) {
                 TurboVisitAction.ADVANCE -> fetchCachedSnapshot()
                 else -> null
             }
 
-            viewLifecycleOwner.lifecycle.whenStateAtLeast(STARTED) {
+            viewLifecycleOwner?.lifecycle?.whenStateAtLeast(STARTED) {
```

Comment:
I, of course, don't understand this nearly as much as you do, but theoretically, isn't there a chance all these guarded calls will create the opposite problem, e.g., navigation stuck, web views not added, etc.?

Reply:
Yes, this is theoretically possible here and something we'll have to watch for in our apps. Since the edge case issues currently lead to crashes, I'd like to try erring on this side which should (theoretically) be recoverable by the user by navigating in the backstack. However, these changes changes should more accurately reflect the difficult reality of timing issues with the message queue and `Fragment` lifecycles.